### PR TITLE
[App Service] `az webapp log tail`: Fix #27189 catch exception when scm connection is lost 

### LINKED ISSUE #27189: az webapp log tail stops after a few minutes
### Describe the bug

I use Azure Linux App Services on a linux app service plan. 

In a WSL Ubuntu Bash shell in Windows Terminal I do:

az webapp log tail --name <name-of-app-service> --resource-group <name-of-resource-group> --debug 

It tails the live log stream for a couple of minutes and then just stops.

When I logstream via: https:<name-of-app-service>.scm.azurewebsites.net/logstream
it stays running (up to the default 200 minutes timeout) as expected.

To make things even more curious. As long as I keep the scm logstream browser active, the az webapp log tail keeps runing. When I stop the browser then quite soon the az webapp log tail stops.

To me this sounds like the scm logstream keeps a process alive that the az webapp log tail also uses??? Really strange this "entanglement".

### Related command

az webapp log tail --name <name-of-app-service> --resource-group <name-of-resource-group> --debug 

### Errors

The tail just stops. No info, not even with --debug

### Issue script & Debug output

This I will show online. The az webapp log tail ... --debug output (blue in the windows terminal) contains a lot of info that I can't share.

### Expected behavior

That it continues to tail the logstream until the configured (or default) timeout.

### Environment Summary

azure-cli                         2.51.0

core                              2.51.0
telemetry                          1.1.0

Dependencies:
msal                            1.24.0b1
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\hvoerman\.azure\cliextensions'

Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:05:00) [MSC v.1929 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

Your CLI is up-to-date.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sarsharma**
**Related command**

`az webapp log tail`

**Description**

While log streaming for a web app, if the connection to the scm logstream endpoint is lost, we currently don't catch the exception properly. This results in the user seeing this error:

```
cli.azure.cli.command_modules.appservice.custom: 2023-08-23T15:39:34 No new trace in the past 1 min(s).
cli.azure.cli.command_modules.appservice.custom: 2023-08-23T15:40:34 No new trace in the past 2 min(s).
cli.azure.cli.command_modules.appservice.custom: 2023-08-23T15:41:34 No new trace in the past 3 min(s).
cli.azure.cli.command_modules.appservice.custom: 2023-08-23T15:42:34 No new trace in the past 4 min(s).
Exception in thread Thread-1 (_get_log):
Traceback (most recent call last):
File "D:\a_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/response.py", line 761, in _update_chunk_length
ValueError: invalid literal for int() with base 16: b''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "D:\a_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/response.py", line 444, in _error_catcher
File "D:\a_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/response.py", line 828, in read_chunked
File "D:\a_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\urllib3/response.py", line 765, in _update_chunk_length
urllib3.exceptions.InvalidChunkLength: InvalidChunkLength(got length b'', 0 bytes read)
```

Wrapping the log stream in a try block to catch any exceptions during logstream.
image

**Testing Guide**

While `az webapp log tail`, if we lose connection to the scm site (eg. the site is restarted)
**History Notes**

[App Service] `az webapp log tail`: Fix #27189 catch exception when scm connection is lost

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2911,14 +2911,18 @@ def _get_log(url, headers, log_file=None):
...
+        try:
+            for chunk in r.stream():
+                if chunk:
+                    # Extra encode() and decode for stdout which does not support 'utf-8'
+                    logger.warning(chunk.decode(encoding='utf-8', errors='replace')
+                                   .encode(std_encoding, errors='replace')
+                                   .decode(std_encoding, errors='replace')
+                                   .rstrip('\n\r'))  # each line of log has CRLF.
+        except:  # pylint: disable=bare-except
+            logger.error("Log stream interrupted. Exiting live log stream.")
```
Comment (vivekjilla) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2923]: let's see if we can log exception also when verbose is enabled.

**Review by vivekjilla — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (vivekjilla) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2923]: oh, just realized we added that too.

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by sarsharma** [comment]
CI test failure is unrelated to this change. Can you please review the PR @zhoxing-ms

**Comment by zhoxing-ms** [comment]
@sarsharma This is a know issue, this CI issue has been fixed by container app team, I will try to re-run the pipeline for you

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**